x-pack/legacy/plugins/apm/public/components/shared/Links/url_helpers.ts

```diff
@@ -4,19 +4,20 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import qs from 'querystring';
+import { parse, stringify } from 'query-string';
 import { LocalUIFilterName } from '../../../../server/lib/ui_filters/local_ui_filters/config';
+import { url } from '../../../../../../../../src/plugins/kibana_utils/public';
 
 export function toQuery(search?: string): APMQueryParamsRaw {
-  return search ? qs.parse(search.slice(1)) : {};
+  return search ? parse(search.slice(1), { sort: false }) : {};
 }
 
 export function fromQuery(query: Record<string, any>) {
-  return qs.stringify(query, undefined, undefined, {
-    encodeURIComponent: (value: string) => {
-      return encodeURIComponent(value).replace(/%3A/g, ':');
-    }
-  });
+  const encodedQuery = url.encodeQuery(query, value =>
+    encodeURIComponent(value).replace(/%3A/g, ':')
+  );
```

Comment:
Thanks for making the changes above!

I think that perhaps `url.encodeQuery` changes the format of the encoded values slightly. This is probably not a problem, but I'd like the change to be explicit.

I've added a few specs to capture previous behaviour. Can you do me a favour and add them to `x-pack/legacy/plugins/apm/public/components/shared/Links/url_helpers.test.tsx`:

```ts
  it('should not encode the following characters', () => {
    expect(
      fromQuery({
        a: true,
        b: 5000,
        c: ':'
      })
    ).toEqual('a=true&b=5000&c=:');
  });

  it('should encode the following characters', () => {
    expect(
      fromQuery({
        a: '@',
        b: '.',
        c: ';',
        d: ' '
      })
    ).toEqual('a=%40&b=.&c=%3B&d=%20');
  });

  it('should handle null and undefined', () => {
    expect(
      fromQuery({
        a: undefined,
        b: null
      })
    ).toEqual('a=&b=');
  });

  it('should handle arrays', () => {
    expect(
      fromQuery({
        arr: ['a', 'b']
      })
    ).toEqual('arr=a&arr=b');
  });
```

Most notably I think the new format changes the array format, and perhaps the handling of `null` and `undefined`.

Reply:
tests were added